assets/js/data/shared-controls.ts

```diff
@@ -48,19 +59,23 @@ export const controls = {
 								response,
 								headers: fetchResponse.headers,
 							} );
+							// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+							// @ts-ignore -- this does exist but doesn't appear to be typed in the api-fetch types.
 							triggerFetch.setNonce( fetchResponse.headers );
 						} )
 						.catch( () => {
 							reject( invalidJsonError );
 						} );
 				} )
 				.catch( ( errorResponse ) => {
+					// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+					// @ts-ignore -- this does exist but doesn't appear to be typed in the api-fetch types.
```

Comment:
Maybe it is because we're adding this method to apiFetch? https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/b8f1cd87ea21715cb5f334ef63a38e689f73c1d6/assets/js/middleware/store-api-nonce.js#L57

Not sure how we can indicate that we're extending `apiFetch` ourselves though so TS knows about it. I found this: https://stackoverflow.com/questions/38971984/how-to-add-custom-typings-in-typescript-2-0-3-0 but it doesn't seem to let me extend `@wordpress/api-fetch` in a way that stops TS showing an error... something to look at when we know a bit more about this maybe?

Reply:
ohhh ya! That would be it :| - We might be able to declare the module in a `d.ts` file to override/enhance (is that what you tried?).

Reply:
Yep that's what I tried, I spent an hour on it last night and another hour on it now and couldn't get it to work. I think this is called module augmentation if you want to have a go at it yourself.

Reply:
I'm going to leave this to work on during the meetup. It can be followed up after this pr.